examples/fetch.rs

```diff
@@ -0,0 +1,194 @@
+// `error_chain!` can recurse deeply
+#![recursion_limit = "1024"]
+
+#[macro_use]
+extern crate error_chain;
+extern crate serde_json;
+extern crate reqwest;
+
+use std::io::{self, Write};
+use std::{env, process};
+
+use serde_json::{Value, map};
+
+// We'll put our errors in a `errors` module
+mod errors {
+    // Create the Error, ErrorKind, ResultExt, and Result types
+    #[allow(unused_doc_comment)]
+    error_chain!{
+        // Automatic conversions between this error chain and other
+        // error types not defined by the `error_chain!`.
+        foreign_links {
+            IO(::std::io::Error);
+            Reqwest(::reqwest::Error);
+        }
+    }
+}
+
+type Map = map::Map<String, Value>;
+
+fn download<F>(uri: &str, mut action: F, debug: bool) -> errors::Result<()>
+    where F: FnMut(Map) -> errors::Result<()> {
+    let json: Value = reqwest::get(uri)?.json()?;
+    let json = if let Value::Object(m) = json {
+        m
+    } else {
+        bail!("Malformed JSON: {:?}", json)
+    };
+
+    if debug {
+        writeln!(io::stderr(), "#json == {}", json.len())?;
+        writeln!(io::stderr(), "License list version {}", get(&json, "licenseListVersion")?)?;
+    }
+    action(json)
+}
+
+fn get<'a>(m: &'a Map, k: &str) -> errors::Result<&'a Value> {
+    if let Some(v) = m.get(k) {
+        Ok(v)
+    } else {
+        bail!("Malformed JSON: {:?} lacks {}", m, k)
+    }
+}
+
+fn main1(args: &[String]) -> errors::Result<()> {
+    let mut upstream_tag = "master";
+    let mut debug = false;
+    for e in args {
+        match e.as_str() {
+            "-d" => {
+                debug = true;
+            },
+            s if s.starts_with('v') && s.ends_with(|c: char| c.is_ascii_digit()) => {
```

Comment:
Can we drop `&& s.ends_with(|c: char| c.is_ascii_digit())` here?  It's breaking `cargo build --examples` on my 1.23.0-dev `rustc` with:

```
error: use of unstable library feature 'ascii_ctype' (see issue #39658)
  --> examples/fetch.rs:62:64
   |
62 |             s if s.starts_with('v') && s.ends_with(|c: char| c.is_ascii_digit()) => {
   |                                                                ^^^^^^^^^^^^^^
   |
   = help: add #![feature(ascii_ctype)] to the crate attributes to enable

error: aborting due to previous error

error: Could not compile `license-exprs`.
```

And I think matching on the opening `v` is a sufficient indicator of version-ness, even without the trailing-digit check.